### PR TITLE
Add pre-cached dotnet6 and remove dotnet 2 from vso focal image

### PR DESCRIPTION
**Pull request by william-msft**
- [x] The purpose of this PR is explained in this message or in an issue. If an issue please include a reference as https://devdiv.visualstudio.com/DevDiv/_boards/board/t/Oryx/Stories/?workitem=1434590
- [ ] Tests are included and/or updated for code changes.
- [ ] Proper license headers are included in each file.

### CODE REVIEW

File: images/build/Dockerfiles/vso.focal.Dockerfile [hunk 1]
```diff
@@ -106,10 +106,10 @@ RUN set -ex \
     && DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1 \
 	&& NUGET_PACKAGES="$nugetPackagesDir" \
     && . $buildDir/__dotNetCoreSdkVersions.sh \
-    && DOTNET_SDK_VER=$DOT_NET_CORE_21_SDK_VERSION \
+    && DOTNET_SDK_VER=$DOT_NET_CORE_31_SDK_VERSION \
        INSTALL_PACKAGES="true" \
        $imagesDir/build/installDotNetCore.sh \
-    && DOTNET_SDK_VER=$DOT_NET_CORE_31_SDK_VERSION \
+       && DOTNET_SDK_VER=DOT_NET_60_SDK_VERSION \
```
Comment (cormacpayne) [inline images/build/Dockerfiles/vso.focal.Dockerfile:112]: Do you need to provide the `$` character at the beginning of `DOT_NET_60_SDK_VERSION`?

Same comment for the change down below as well.

Comment (william-msft) [reply images/build/Dockerfiles/vso.focal.Dockerfile]: Thanks @cormacpayne ! I've updated

File: images/build/Dockerfiles/vso.focal.Dockerfile [hunk 2]
```diff
@@ -106,10 +106,10 @@ RUN set -ex \
     && DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1 \
 	&& NUGET_PACKAGES="$nugetPackagesDir" \
     && . $buildDir/__dotNetCoreSdkVersions.sh \
-    && DOTNET_SDK_VER=$DOT_NET_CORE_21_SDK_VERSION \
+    && DOTNET_SDK_VER=$DOT_NET_CORE_31_SDK_VERSION \
        INSTALL_PACKAGES="true" \
        $imagesDir/build/installDotNetCore.sh \
-    && DOTNET_SDK_VER=$DOT_NET_CORE_31_SDK_VERSION \
+       && DOTNET_SDK_VER=$DOT_NET_60_SDK_VERSION \
```
Comment (cormacpayne) [inline images/build/Dockerfiles/vso.focal.Dockerfile:112]: nit: the alignment of this line to the others that start with `&&` is now off -- would you mind lining it back up?

**Review by cormacpayne — APPROVED** [review]
Some tests are failing because they haven't been updated to remove the check for .NET 2.1 -- once those are updated, this should be good to merge (we should also add tests for checking .NET 6 is present)

**Review by gpcastro — APPROVED** [review]

**Pull request merged**